src/app/login/login.component.ts

```diff
@@ -0,0 +1,133 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {ComponentsModule} from "@biggive/components-angular";
+import {MatButtonModule} from "@angular/material/button";
+import {MatDialogModule} from "@angular/material/dialog";
+import {MatFormFieldModule} from "@angular/material/form-field";
+import {MatInputModule} from "@angular/material/input";
+import {MatProgressSpinnerModule} from "@angular/material/progress-spinner";
+import {FormBuilder, FormGroup, ReactiveFormsModule, Validators} from "@angular/forms";
+import {RecaptchaComponent, RecaptchaModule} from "ng-recaptcha";
+import {Credentials} from "../credentials.model";
+import {IdentityService} from "../identity.service";
+import {environment} from "../../environments/environment";
+import {EMAIL_REGEXP} from "../validators/patterns";
+import {ActivatedRoute, Router} from "@angular/router";
+import {MatSnackBar, MatSnackBarModule} from "@angular/material/snack-bar";
+
+@Component({
+  selector: 'app-login',
+  standalone: true,
+  imports: [CommonModule, ComponentsModule, MatButtonModule, MatDialogModule, MatFormFieldModule, MatInputModule, MatProgressSpinnerModule, ReactiveFormsModule, RecaptchaModule, MatSnackBarModule],
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit{
+  @ViewChild('captcha') captcha: RecaptchaComponent;
+  protected forgotPassword = false;
+  protected loggingIn = false;
+  protected loginError?: string;
+  loginForm: FormGroup;
+  protected userAskedForResetLink = false;
+  protected resetPasswordForm: FormGroup;
+  protected resetPasswordSuccess: boolean|undefined = undefined;
+  protected recaptchaIdSiteKey = environment.recaptchaIdentitySiteKey;
+  private targetUrl: URL = new URL(environment.donateGlobalUriPrefix);
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private identityService: IdentityService,
+    private activatedRoute: ActivatedRoute,
+  ) {
+  }
+
+  ngOnInit() {
+    this.loginForm = this.formBuilder.group({
+      emailAddress: [null, [
+        Validators.required,
+        Validators.pattern(EMAIL_REGEXP),
+      ]],
+      password: [null, [
+        Validators.required,
+        Validators.minLength(10),
```

Comment:
not sure it's worth having a minLength validation on the password field here. It's here because I copied this from the login modal. Min length is good when setting a new password, but I think it might be more confusing than helpful when trying to authenticate with an existing password.

The only password validation rule as far as the user is concerned here is that the password has to match what's set on the account. Yes we know that that implies it will be at least 10 characters, but that feels a bit arbitrary to enforce here.

Reply:
I think you're right, I don't remember being notified about password length on any login page, but maybe it's good to check length at the form level before checking against db, as we know it can't be right and won't make unnecessary db calls?

Reply:
I think if we do that for performance reasons then we don't need to tell the user about it, as it just makes things more complicated for them - and if we do it it makes more sense to do it in the Identity server instead of in frontend. I'll leave it here for now but likely change in a later PR.